Detectors/ITSMFT/ITS/tracking/src/Tracker.cxx

```diff
@@ -412,17 +411,34 @@ bool Tracker::fitTrack(const ROframe& event, TrackITSExt& track, int start, int
     if (!track.propagateTo(trackingHit.xTrackingFrame, getBz())) {
       return false;
     }
-
-    track.setChi2(track.getChi2() +
-                  track.getPredictedChi2(trackingHit.positionTrackingFrame, trackingHit.covarianceTrackingFrame));
+    auto predChi2{track.getPredictedChi2(trackingHit.positionTrackingFrame, trackingHit.covarianceTrackingFrame)};
+    if (predChi2 > chi2cut) {
+      return false;
+    }
+    track.setChi2(track.getChi2() + predChi2);
     if (!track.o2::track::TrackParCov::update(trackingHit.positionTrackingFrame, trackingHit.covarianceTrackingFrame)) {
       return false;
     }
 
-    const float xx0 = (iLayer > 2) ? 0.008f : 0.003f; // Rough layer thickness
-    constexpr float radiationLength = 9.36f;          // Radiation length of Si [cm]
-    constexpr float density = 2.33f;                  // Density of Si [g/cm^3]
-    if (!track.correctForMaterial(xx0, xx0 * radiationLength * density, true)) {
+    float xx0 = ((iLayer > 2) ? 0.008f : 0.003f); // Rough layer thickness
+    float radiationLength = 9.36f;                // Radiation length of Si [cm]
+    float density = 2.33f;                        // Density of Si [g/cm^3]
+    float distance = xx0;                         // Default thickness
+
+    if (mMatLayerCylSet) {
+      if ((iLayer + step) != end) {
+        const auto cl_0 = mPrimaryVertexContext->getClusters()[iLayer][track.getClusterIndex(iLayer)];
+        const auto cl_1 = mPrimaryVertexContext->getClusters()[iLayer + step][track.getClusterIndex(iLayer + step)];
+
+        distance = std::hypot(cl_0.xCoordinate - cl_1.xCoordinate, cl_0.yCoordinate - cl_1.yCoordinate, cl_0.zCoordinate - cl_1.zCoordinate);
```

Comment:
note that the ``matbud`` provides the distance between 2 points, you don't need to calculate it, just use
````
distance = matbud.length;
````